File: apps/ios/Pebbles/Features/Path/Render/PebbleStroke.swift
import CoreGraphics

/// Stroke geometry for the pebble layer-tracing renderers.
///
/// The composed pebble SVG authors the outline (`layer:shape`) at
/// `stroke-width="6"` in viewBox units, but the glyph (`layer:glyph`) is
/// authored thinner (`6 × zoneScale`). Tracing every layer at the outline's
/// weight makes the glyph read the same weight as the outline — the fix for
/// issue #509.
enum PebbleStroke {

    /// The outline's authored stroke width in viewBox units. Mirrors the
    /// engine's shape/glyph stored stroke (`GLYPH_STROKE_WIDTH`, and the `6`
    /// on every shape path in the engine's shape templates).
    static let outlineWidth: CGFloat = 6

    /// On-screen `lineWidth` for a layer traced into `frame`, proportional to
    /// how the viewBox fits the frame (uniform min-axis scale, matching
    /// `LayerShape`'s fit). Keeps the stroke's absolute weight consistent with
    /// the outline at any thumbnail size.
    static func lineWidth(viewBox: CGRect, frame: CGSize) -> CGFloat {
        guard viewBox.width > 0, viewBox.height > 0 else { return outlineWidth }
        let scale = min(frame.width / viewBox.width, frame.height / viewBox.height)
        return outlineWidth * scale
    }
}

File: apps/android/app/src/main/kotlin/app/pbbls/android/features/path/render/PebbleStroke.kt
package app.pbbls.android.features.path.render

/**
 * Stroke geometry for the pebble layer-tracing renderer ([PebbleStaticRender]).
 *
 * The server-composed pebble SVG authors the outline (`layer:shape`) at
 * `stroke-width="6"` in viewBox units, but each glyph is authored at its own
 * width: a custom carved glyph is normalized to `strokeWidth × fitScale` (often
 * heavier), while a domain glyph carries the icon's authored width (often
 * lighter). AndroidSVG honors those per-layer widths, so on the Path and detail
 * pages custom glyphs render thicker than domain glyphs.
 *
 * Tracing every layer at the outline's weight makes glyph == outline
 * everywhere, so custom and domain glyphs read identically — the Android analog
 * of the iOS fix in PR #511.
 */
internal object PebbleStroke {
    /**
     * Outline stroke width in viewBox units — the weight every layer is traced
     * at. Mirrors the `6` on every shape path in the engine's shape templates
     * (and iOS `PebbleStroke.outlineWidth`). The renderer scales the whole
     * viewBox to the frame, so the on-screen weight is `OUTLINE_WIDTH × fitScale`
     * — exactly today's outline weight, now shared by the glyph.
     */
    const val OUTLINE_WIDTH: Float = 6f
}
